.github/workflows/uploadFirtoolBinaries.yml

```diff
@@ -66,6 +66,8 @@ jobs:
               -DCMAKE_C_COMPILER=${{ matrix.cc }} \
               -DCMAKE_CXX_COMPILER=${{ matrix.cxx }} \
               -DBUILD_SHARED_LIBS=${{ matrix.build_config.shared }} \
+              -DLLVM_USE_LINKER=gold \
```

Comment:
Just curious, any specific reason to prefer `gold`? When I've used this in the past, I've actually tended towards `lld`, but curious if there's something nice about `gold`?

Reply:
In my experience (and I could be wrong) gold links a lot faster. Since I'm trying to disable building the tools (and thus linking) in the LLVM build, it may be moot. When I was testing this locally, the linking was extremely slow with the default (which I presume is `lld` but I'm not actually sure), and pretty snappy with `gold`. I'm talking multiple minutes of linking on a relatively fast machine.

Reply:
`gold` is an ELF-only linker so I'm just going to back this change out.

Reply:
The default is almost always ld, which is really slow.  In my experience, ld.lld is usually faster than gold and has the upside that it comes with llvm.

Reply:
Trying out default: https://github.com/jackkoenig/circt/actions/runs/4535349993
and trying out lld: https://github.com/jackkoenig/circt/actions/runs/4535458618

Reply:
I'm just going to leave the linking alone for now.